ddtrace/internal/processor/trace.py

```diff
@@ -95,9 +105,7 @@ def process_trace(self, trace):
 
         span_ids = {span.span_id for span in trace}
         for span in trace:
-            if span is span._local_root:
-                span.set_metric("_dd.top_level", 1)
-            elif span._parent and span.service != span._parent.service:
+            if _is_top_level(span):
                 span.set_metric("_dd.top_level", 1)
             elif span.parent_id and span.parent_id not in span_ids:
```

Comment:
Are there still cases where the agent is expected compute stats? If not then then we could remove this processor

Reply:
yes, if stats computation is not enabled in the client (which is the default)